### PR TITLE
Gradle related items

### PR DESCRIPTION
**Pull request by Josh-Gebbeken**
Completed Gradle items taken from Android Arch Demo

### CODE REVIEW

**Review by BottleRocket-Colin — DISMISSED** [review]
This is a great start.  as we get moving along, we'll want to trim out unused dependencies and verify Jacoco coverage etc.

**Comment by Josh-Gebbeken** [comment]
> This is a great start. as we get moving along, we'll want to trim out unused dependencies and verify Jacoco coverage etc.

Agreed. There might be some extra dependencies that we don't actually need.

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+extra.set("jacocoCoverageThreshold", 0.40.toBigDecimal()) // module specific code coverage verification threshold
+
+apply(from = "../renameAppBundle.gradle.kts") // configures additional gradle tasks to rename app bundles (when needed)
+
+// Prep BuildInfoManager to use its functions/properties later throughout this build script
+BuildInfoManager.initialize(
+    BuildInfoInput(
+        appVersion = AppVersion(major = 1, minor = 0, patch = 0, hotfix = 0, showEmptyPatchNumberInVersionName = true), // TODO: TEMPLATE - Replace with appropriate app version
+        brandName = "BR_Architecture", // TODO: TEMPLATE - Replace with appropriate project brand name
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken Likely rename this to `BR_Maps_Demo` and remove the `// TODO: TEMPLATE` comment

```suggestion
        brandName = "BR_Maps_Demo"
```

File: app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+extra.set("jacocoCoverageThreshold", 0.40.toBigDecimal()) // module specific code coverage verification threshold
+
+apply(from = "../renameAppBundle.gradle.kts") // configures additional gradle tasks to rename app bundles (when needed)
+
+// Prep BuildInfoManager to use its functions/properties later throughout this build script
+BuildInfoManager.initialize(
+    BuildInfoInput(
+        appVersion = AppVersion(major = 1, minor = 0, patch = 0, hotfix = 0, showEmptyPatchNumberInVersionName = true), // TODO: TEMPLATE - Replace with appropriate app version
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken nit: As this is just a demo for maps, you should be able to remove all of the `// TODO: TEMPLATE` comments from the project

File: app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            isMinifyEnabled = true
+            isShrinkResources = true
+            proguardFiles(getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro")
+        }
+    }
+    flavorDimensions += listOf("environment")
+    // See BEST_PRACTICES.md for comments on purpose of each build type/flavor/variant
+    productFlavors {
+        create("internal") {
+            applicationId = "com.bottlerocketstudios.brarchitecture.internal" // TODO: TEMPLATE - Replace with appropriate project applicationId prefix, leaving .internal
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken Rename this to use a unique applicationId for this project, potentially `com.bottlerocketstudios.mapsdemo.internal`

File: app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            versionNameSuffix = "-internal"
+            buildConfigField("boolean", "INTERNAL", "true")
+            buildConfigField("boolean", "PRODUCTION", "false")
+            dimension = "environment"
+        }
+        create("production") {
+            applicationId = "com.bottlerocketstudios.brarchitecture" // TODO: TEMPLATE - Replace full string with appropriate project applicationId
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken Rename this to use a unique applicationId for this project, potentially `com.bottlerocketstudios.mapsdemo`

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.bottlerocketstudios.android_maps_demo">
```
Comment (jaredstockton) [inline app/src/main/AndroidManifest.xml]: @Josh-Gebbeken Suggest using `com.bottlerocketstudios.mapsdemo` to match applicationId

File: app/src/main/java/com/bottlerocketstudios/android_maps_demo/ui/theme/Color.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.bottlerocketstudios.android_maps_demo.ui.theme
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/android_maps_demo/ui/theme/Color.kt]: Refactor all files under `com.bottlerocketstudios.android_maps_demo` to `com.bottlerocketstudios.mapsdemo`

**Review by jaredstockton — CHANGES_REQUESTED** [review]
This is a good start! A few minor things to address, mainly around applicationId and packagenames

File: app/build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    productFlavors {
+        create("internal") {
+            applicationId = "com.bottlerocketstudios.mapsdemo.internal"
+            versionNameSuffix = "-internal"
+            buildConfigField("boolean", "INTERNAL", "true")
+            buildConfigField("boolean", "PRODUCTION", "false")
+            dimension = "environment"
+        }
+        create("production") {
+            applicationId = "com.bottlerocketstudios.mapsdemo.internal"
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken Remove `.internal` from the production applicationId

```suggestion
            applicationId = "com.bottlerocketstudios.mapsdemo"
```

File: app/src/androidTest/java/com/bottlerocketstudios/mapsdemo/ExampleInstrumentedTest.kt [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+    @Test
+    fun useAppContext() {
+        // Context of the app under test.
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        assertEquals("com.example.android_maps_demo", appContext.packageName)
```
Comment (jaredstockton) [inline app/src/androidTest/java/com/bottlerocketstudios/mapsdemo/ExampleInstrumentedTest.kt]: @Josh-Gebbeken nit: Need to refactor this, likely to `com.bottlerocketstudios.mapsdemo`

```suggestion
        assertEquals("com.bottlerocketstudios.mapsdemo", appContext.packageName)
```

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.bottlerocketstudios.mapsdemo.internal">
```
Comment (jaredstockton) [inline app/src/main/AndroidManifest.xml]: @Josh-Gebbeken Refactor this to `com.bottlerocketstudios.mapsdemo`

```suggestion
    package="com.bottlerocketstudios.mapsdemo">
```

**Review by jaredstockton — CHANGES_REQUESTED** [review]
A few more changes requested around the applicationId and packagename

**Review by BottleRocket-Colin — DISMISSED** [review]

File: app/build.gradle.kts [hunk 10]
```diff
@@ -7,15 +7,19 @@ plugins {
...
     id(Config.ApplyPlugins.PARCELIZE)
 }
+// module specific code coverage verification threshold
+extra.set("jacocoCoverageThreshold", 0.40.toBigDecimal())
 
-extra.set("jacocoCoverageThreshold", 0.40.toBigDecimal()) // module specific code coverage verification threshold
-
-apply(from = "../renameAppBundle.gradle.kts") // configures additional gradle tasks to rename app bundles (when needed)
+apply(from = "../renameAppBundle.gradle.kts") // configures additional gradle tasks to rename
+// app bundles (when needed)
```
Comment (jaredstockton) [inline app/build.gradle.kts]: @Josh-Gebbeken nit: It looks like some file-wide code cleanup caused much of this file to be modified (likely due to a lower characters per line default). I'd say undo all of the changes to this file made in this commit and just redo the actual change of applicationId to `com.bottlerocketstudios.mapsdemo` in the release buildtype section as some of them are a little excessive (going from 2 lines for comment and code for the keystore file to 9 lines)

**Review by jaredstockton — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 10 above)
Comment (Josh-Gebbeken) [reply app/build.gradle.kts]: fixed

Comment (jaredstockton) [reply app/build.gradle.kts]: @Josh-Gebbeken After your commits mentioned that ktlint performed these formatting changes, it reminded me of the .editorconfig file which manages max line length. I'd recommend you bring in [the file from the architecture demo](https://github.com/BottleRocketStudios/Android-ArchitectureDemo/blob/develop/.editorconfig) and re-run ktlint (it should respect the line length value of 200 and not auto format this file with an 80 character length limit) :)

**Review by jaredstockton — APPROVED** [review]
Thanks for making all the changes @Josh-Gebbeken!

File: app/build.gradle.kts (hunk 10 above)
Comment (Josh-Gebbeken) [reply app/build.gradle.kts]: File added to project and ktlint formatted and ran with no issues.

**Review by BottleRocket-Colin — APPROVED** [review]

**Pull request merged**